docs/tutorials/embedded/wine_detector.md

```diff
@@ -0,0 +1,386 @@
+streamHandler = logging.StreamHandler()
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+streamHandler.setFormatter(formatter)
+logger.addHandler(streamHandler)
+
+
+# Init AWSIoTMQTTClient For Publish/Subscribe Communication With Server
+myAWSIoTMQTTClient = None
+if useWebsocket:
+    myAWSIoTMQTTClient = AWSIoTMQTTClient("basicPubSub", useWebsocket=True)
+    myAWSIoTMQTTClient.configureEndpoint(host, 443)
+    myAWSIoTMQTTClient.configureCredentials(rootCAPath)
+else:
+    myAWSIoTMQTTClient = AWSIoTMQTTClient("basicPubSub")
+    myAWSIoTMQTTClient.configureEndpoint(host, 8883)
+    myAWSIoTMQTTClient.configureCredentials(rootCAPath, privateKeyPath, certificatePath)
+
+
+# AWSIoTMQTTClient connection configuration
+myAWSIoTMQTTClient.configureAutoReconnectBackoffTime(1, 32, 20)
+myAWSIoTMQTTClient.configureOfflinePublishQueueing(-1)  # Infinite offline Publish queueing
+myAWSIoTMQTTClient.configureDrainingFrequency(2)  # Draining: 2 Hz
+myAWSIoTMQTTClient.configureConnectDisconnectTimeout(10)  # 10 sec
+myAWSIoTMQTTClient.configureMQTTOperationTimeout(5)  # 5 sec
+
+
+# Connect and subscribe to AWS IoT
+myAWSIoTMQTTClient.connect()
+myAWSIoTMQTTClient.subscribe("sdk/test/Python", 1, customCallback)
+time.sleep(2)
+
+
+# Start the Camera and tell the Server we are alive
+print "Running camera"
+myAWSIoTMQTTClient.publish("sdk/test/Python", "New Message: Starting Camera", 0)
+camera = picamera.PiCamera()
+
+# Capture forever (this is a modified version of camera_test.py)
+while True:
+    filename = '/home/pi/cap.jpg'
+    camera.start_preview()
+    camera.capture(filename)
+    camera.stop_preview()
+    topn = inception_predict.predict_from_local_file(filename, N=5)
+    
+    # Check if either of the top two predictions are wine related and publish a message if it is
+    # you can change 'wine' here to anything you want to alert the server about detecting
+    if 'wine' in topn[0][1] or 'wine' in topn[1][1]: 
+        myAWSIoTMQTTClient.publish("sdk/test/Python", "New Message: WINE DETECTED!", 0)
+```
+
+You can then run this file by entring the following command
+
+```bash
+python wine_alerter.py -e <endpointURL> -r <rootCAFilePath> -c <certFilePath> -k <privateKeyFilePath>
+```
+
+If this is working you should see the same kind of image preview you did with camera_test.py every few seconds, however the console will only print a message now when a wine bottle is detected in the shot (you can edit the bottom lines in the wine_alerter.py code to make this alert for any object you specify).
+
+You can open up the activity tab for the thing you registered on the AWS IoT Dashboard and see the corresponding messages pushed to the server whenever a wine bottle is detected in shot. Thanks to AWS IoT even if network connectivity periodically fails, the updates will eventually be pushed out to the server when possible, allowing this system to reliably let you know when there is wine around.
```

Comment:
Suggested word tweaking:
You can open up the activity tab for the thing **that** you registered on the AWS IoT Dashboard and see the corresponding messages pushed to the server whenever a wine bottle is detected in **a camera** shot. **Even if network connectivity periodically fails, AWS IoT will push updates out to the server when possible,** allowing this system to reliably let you know when there is wine around.